src/rendering/luaconsole.cpp

```diff
@@ -893,4 +674,517 @@ void LuaConsole::parallelConnectionChanged(const ParallelConnection::Status& sta
+    else {
+        // Not a valid dir - check the parent
+        std::filesystem::path parent = path.parent_path();
+        if (std::filesystem::exists(parent) && std::filesystem::is_directory(parent)) {
+            dirToSearch = parent;
+        }
+        else {
+            // Neither path nor parent is valid, cancel autocomplete
+            return false;
+        }
+    }
+
+    // Get the entries in directory
+    std::vector<std::filesystem::path> suggestions =
+        ghoul::filesystem::walkDirectory(
+            dirToSearch,
+            ghoul::filesystem::Recursive::No,
+            ghoul::filesystem::Sorted::Yes
+        );
+
+    auto containsNonAscii = [](const std::filesystem::path& p) {
+        auto s = p.generic_u8string();
+        for (auto it = s.rbegin(); it != s.rend(); it++) {
+            if (static_cast<unsigned char>(*it) > 0x7F) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    std::vector<std::string> entries;
+    for (const auto& entry : suggestions) {
+        // Filter paths that contain non-ASCII characters
+        if (containsNonAscii(entry)) {
+            continue;
+        }
+        
+        entries.push_back(entry.string());
+    }
+
+    _autoCompleteState.suggestions = std::move(entries);
+    _autoCompleteState.input = userTypedPath;
+
+    if (pathEnd != std::string::npos) {
+        // There is something after the path so we want to insert inbetween
+        _autoCompleteState.insertPosition = contextStart + pathEnd;
+    }
+    else {
+        // There is nothing after the path so we render normally at the end
+        _autoCompleteState.insertPosition = currentCommand.size();
+    }
+
+    return true;
+}
+
+void LuaConsole::gatherFunctionSuggestions(size_t contextStart) {
+    const std::string currentCommand = _commands[_activeCommand];
+    // Get a list of all the available commands
+    std::vector<std::string> allCommands = global::scriptEngine->allLuaFunctions();
+    std::sort(allCommands.begin(), allCommands.end());
+
+    _autoCompleteState.suggestions = std::move(allCommands);
+
+    std::string possibleFunction = currentCommand.substr(contextStart);
+    // Find the nearest parenthesis if any exists, which marks the end of the
+    // function. Otherwise the rest of the string is assumed to be part of
+    // the function
+    size_t functionEnd = possibleFunction.find_first_of("()");
+    _autoCompleteState.input = possibleFunction.substr(0, functionEnd);
+
+    if (functionEnd != std::string::npos) {
+        // There is something after the function so we want to insert inbetween
+        _autoCompleteState.insertPosition = contextStart + functionEnd;
+    }
+    else {
+        // There is nothing after the path so we render normally at the end
+        _autoCompleteState.insertPosition = currentCommand.size();
+    }
+}
+
+void LuaConsole::filterSuggestions() {
+    auto normalize = [this](const std::string& s) -> std::string {
```

Comment:
```suggestion
    auto normalize = [this](const std::string& s) {
```

If it is not necesasry, we normally don't explicitly specify the return value for lambdas